neo/Network/RPC/TransactionManager.cs

```diff
@@ -0,0 +1,220 @@
+﻿using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Network.P2P.Payloads;
+using Neo.Network.RPC.Models;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.VM;
+using Neo.Wallets;
+using System;
+
+namespace Neo.Network.RPC
+{
+    /// <summary>
+    /// This class helps to create transaction with RPC API.
+    /// </summary>
+    public class TransactionManager
+    {
+        private static readonly Random rand = new Random();
+        private readonly RpcClient rpcClient;
+        private readonly UInt160 sender;
+
+        /// <summary>
+        /// The Transaction context to manage the witnesses
+        /// </summary>
+        private ContractParametersContext context;
+
+        /// <summary>
+        /// The Transaction managed by this class
+        /// </summary>
+        public Transaction Tx { get; private set; }
+
+        /// <summary>
+        /// TransactionManager Constructor
+        /// </summary>
+        /// <param name="rpc">the RPC client to call NEO RPC API</param>
+        /// <param name="sender">the account script hash of sender</param>
+        public TransactionManager(RpcClient rpc, UInt160 sender)
+        {
+            rpcClient = rpc;
+            this.sender = sender;
+        }
+
+        /// <summary>
+        /// Create an unsigned Transaction object with given parameters.
+        /// </summary>
+        /// <param name="script">Transaction Script</param>
+        /// <param name="attributes">Transaction Attributes</param>
+        /// <param name="cosigners">Transaction Cosigners</param>
+        /// <param name="networkFee">Transaction NetworkFee, will set to estimate value(with only basic signatures) when networkFee is 0</param>
+        /// <returns></returns>
+        public TransactionManager MakeTransaction(byte[] script, TransactionAttribute[] attributes = null, Cosigner[] cosigners = null, long networkFee = 0)
```

Comment:
Is not possible to reuse `Wallet.MakeTransactions` this method already compute the fee

Reply:
Hi Shargon
It would be perfect if we can reuse ```Wallet.MakeTransactions```, but we can't use snapshot and ApplicationEngine in RPC SDK. So we have extracted the``` CalculateNetWorkFee``` method and the system fee is calculated by RPC.